license.Rmd

```diff
@@ -39,31 +40,33 @@ packages <- as_tibble(available.packages())
 parsed <- packages %>% 
   select(package = Package, license = License) %>% 
   mutate(
-    or_file = str_detect(license, fixed("| file LICENSE")),
-    plus_file = str_detect(license, fixed("+ file LICENSE")),
-    license = str_remove(license, " [+|] file LICENSE")
+    or_file = str_detect(license, fixed("| file LICEN[CS]E")),
+    plus_file = str_detect(license, fixed("+ file LICEN[CS]E")),
+    license = str_remove(license, " [+|] file LICEN[CS]E")
   )
 
 parsed %>% count(license, sort = TRUE)
+nrow(parsed)
 
-parsed %>% 
+copyleft <- parsed %>% 
   filter(str_detect(license, "GPL")) %>% 
-  count(license, sort = TRUE) %>% 
-  head(10) %>% 
-  knitr::kable()
+  filter(!str_detect(license, "LGPL")) %>% 
+  count(license, sort = TRUE)
+copyleft
+sum(copyleft$n) / nrow(parsed)
 
-parsed %>% 
-  filter(!str_detect(license, "GPL")) %>% 
+permissive <- parsed %>% 
   count(license, sort = TRUE) %>% 
-  head(10) %>% 
-  knitr::kable()
+  anti_join(copyleft) %>% 
+  filter(license %in% c("MIT", "CC0") | str_detect(license, "BSD"))
+permissive 
+sum(permissive$n) / nrow(parsed)
 ```
 
-(If you run the code in this chapter, please make sure that you're using usethis 2.0.0 or greater; writing this chapter prompted a number of changes in the package.)
-
 ## Code you write
 
-We'll start by talking about code that you write, and how to license it to make clear how you want people to treat it.
+We'll start by talking about code that you write, and how to license it to make clear how you want people to treat it/ It's important to use a license because if you don't the default copyright laws apply which mean that no one is allowed to make a copy of your code without your express permission.
```

Comment:
```suggestion
We'll start by talking about code that you write, and how to license it to make clear how you want people to treat it.
It's important to use a license because if you don't, the default copyright laws apply, which means that no one is allowed to make a copy of your code without your express permission.
```